Jobsway2goMvc/Areas/Identity/Pages/Account/Register.cshtml.cs

```diff
@@ -123,52 +129,63 @@ public async Task<IActionResult> OnPostAsync(string returnUrl = null)
             ExternalLogins = (await _signInManager.GetExternalAuthenticationSchemesAsync()).ToList();
             if (ModelState.IsValid)
             {
-                var user = CreateUser();
-                user.FirstName = Input.FirstName;
-                user.LastName = Input.LastName;
-                await _userStore.SetUserNameAsync(user, Input.Email, CancellationToken.None);
-                await _emailStore.SetEmailAsync(user, Input.Email, CancellationToken.None);
-               
-
-                var userRole = _roleManager.FindByNameAsync("User").Result;
-                if (userRole == null)
-                {
-                    ViewData["UserRoleIsNull"] = "Something wrong happened";
-                    return Page();
-                }
-                var result = await _userManager.CreateAsync(user, Input.Password);
-                if (result.Succeeded)
+                var validateUser = new RegistrationValidator();
+                FluentValidation.Results.ValidationResult result1 = await validateUser.ValidateAsync(Input);
+                if (result1.IsValid)
```

Comment:
Rename from `validateUser` to `validator` and from `result1` to `validationResult`. You can also make the validationResult as `var` instead of `FluentValidation.Results.ValidationResult`